### DISCUSSION TITLE
Get the file names dui

### DISCUSSION THREAD

**Discussion by qngyn** [quarto-dev/quarto-cli#10443] [Q&A]
### Description

Hi, 
I wonder if there is any way I can try to get the name of each quarto file as it's being processed?  I want to attach an audio file to the quarto file where the audio file's base name matches the Quarto file's name.

**Comment by mcanouil** [comment]
Would you mind sharing a more concrete example of what you are trying to achieve?

**Reply by qngyn** [reply]
Hi,
So, I am trying to attach a list of audio files, and each file is specific for each quarto file. So, I believe in achieving that, I should somehow be able to get the quarto file name while rendering the quarto book and then modify the content (adding the audio). However,  I am struggling to find a way to get the quarto file names during rendering

**Reply by mcanouil** [reply]
Are you generating the Quarto documents?
If not, why not hardcoding the audio file name?

You seem to want something that is way more complicated than what's needed.
Anyhow, if you think the hard way is the way, then you can write a Lua filter/shortcode that will add the relevant audio file using the proper code such as done in the example of https://github.com/mcanouil/quarto-lua-env which shows the Lua variables containing the input file.
You could possibly use the pre/post render scripts and the environment variables to get the input files: https://quarto.org/docs/projects/scripts.html#pre-and-post-render

But again, this feels way more complicated than it needs to be so I might be missing something in your use case.

**Reply by qngyn** [reply]
the audio's name will be hardcoded but I think the links you provided would help! Thank you a lot
